packages/web-app-external/src/App.vue

```diff
@@ -1,160 +1,150 @@
   name: 'ExternalApp',
-  components: {
-    AppTopBar,
-    ErrorScreen,
-    LoadingScreen
+  props: {
+    resource: { type: Object as PropType<Resource>, required: true }
   },
-  setup() {
+  setup(props) {
+    const language = useGettext()
+    const store = useStore()
+
+    const { $gettext, interpolate: $gettextInterpolate } = language
+    const { makeRequest } = useRequest()
+
     const appName = useRouteQuery('app')
+    const appUrl = ref()
+    const formParameters = ref({})
+    const method = ref()
+    const subm: VNodeRef = ref()
+
+    const capabilities = computed(() => store.getters['capabilities'])
+    // FIXME: Make available to appDefaults composable for pageTitle
     const applicationName = computed(() => queryItemAsString(unref(appName)))
-    return {
-      ...useAppDefaults({
-        applicationId: 'external',
-        applicationName
-      }),
-      applicationName
-    }
-  },
 
-  data: () => ({
-    appUrl: '',
-    errorMessage: '',
-    formParameters: {},
-    loading: false,
-    loadingError: false,
-    method: '',
-    resource: null
-  }),
-  computed: {
-    ...mapGetters(['capabilities']),
-
-    pageTitle() {
-      const translated = this.$gettext('"%{appName}" app page')
-      return this.$gettextInterpolate(translated, {
-        appName: this.applicationName
+    const iFrameTitle = computed(() => {
+      const translated = $gettext('"%{appName}" app content area')
+      return $gettextInterpolate(translated, {
+        appName: unref(applicationName)
       })
-    },
-    iFrameTitle() {
-      const translated = this.$gettext('"%{appName}" app content area')
-      return this.$gettextInterpolate(translated, {
-        appName: this.applicationName
+    })
+
+    // TODO: Extract within web-pkg with default title?
+    const errorPopup = (error) => {
+      store.dispatch('showErrorMessage', {
+        title: $gettext('An error occurred'),
+        desc: error,
+        error
       })
-    },
-    fileIdFromRoute() {
-      return this.$route.query.fileId
     }
-  },
-  async created() {
-    this.loading = true
-    try {
-      this.resource = await this.getFileInfo(this.currentFileContext, {
-        davProperties: []
-      })
 
-      const fileId = this.fileIdFromRoute || this.resource.fileId
-
-      // fetch iframe params for app and file
-      const baseUrl = urlJoin(
-        configurationManager.serverUrl,
-        this.capabilities.files.app_providers[0].open_url
-      )
-      const query = stringify({
-        file_id: fileId,
-        lang: this.$language.current,
-        ...(this.applicationName && { app_name: this.applicationName })
-      })
-      const url = `${baseUrl}?${query}`
-      const response = await this.makeRequest('POST', url, {
-        validateStatus: () => true
-      })
+    const loadAppUrl = useTask(function* () {
+      try {
+        // FIXME: make all updates atomic/happen at once so the ui can never mix data from different resources
```

Comment:
probably invalid comment, we show an error anyway when something goes wrong

Reply:
We should instead make sure to go into an error state...

Reply:
Marking as resolved